openmdao/core/component.py

```diff
@@ -1702,10 +1702,13 @@ def _check_consistent_serial_dinputs(self, nz_dist_outputs):
 
         if self._serial_idxs is None:
             ranges = defaultdict(list)
+            globaloffset = 0
             for name, offset, end, vec, slc, dist_sizes in self._jac_wrt_iter():
+                if name.split(".")[-1] in self._var_rel_names["output"]:
```

Comment:
could replace this with `if vec is self._outputs:` here.

Reply:
Is this the correct approach overall? I noticed below that it selects either the outputs or the inputs, so would this fix conflict with a case in which the outputs are checked instead of the inputs?

Reply:
Actually, you're right.  Maybe a better way to do this would be:
```
output_len = len(self._outputs)
for name, offset, end, vec, slc, dist_sizes in self._jac_wrt_iter():
    if dist_sizes is None:  # not distributed
        if offset != end:
            if vec is self._outputs:
                ranges[vec].append(range(offset, end))
            else:
                ranges[vec].append(range(offset - output_len, end - output_len))
```

I didn't test it, but I think this should work because `_jac_wrt_iter` yields all of the outputs first, then the inputs (for an implicit component).

Reply:
I tried this out but it fails for explicit components since the `_jac_wrt_iter()` object does not include the outputs in that case. I added a loop to check if there are outputs before setting the offset. My fix still feels like a bit of a hack, but it seems to work.